unumpy/linalg/_multimethods.py

```diff
@@ -0,0 +1,224 @@
+import functools
+import operator
+from uarray import create_multimethod, mark_as, all_of_type, Dispatchable
+import builtins
+
+create_numpy = functools.partial(create_multimethod, domain="numpy.linalg")
+
+from .._multimethods import (
+    ndarray,
+    _identity_argreplacer,
+    _self_argreplacer,
+    _dtype_argreplacer,
+    mark_dtype,
+    _first_argreplacer,
+    _first2argreplacer,
+)
+
+
+@create_numpy(_first_argreplacer)
```

Comment:
I'd add a default here, it's easy:

```
ret = arrays[0]
for a in arrays[1:]:
    ret = ret @ a

return ret
```

Reply:
One can do a more sophisticated algorithm depending on the shapes, but it's okay for now.

Reply:
Could we write `multi_dot_default` with respect to `dot`?

A simple implementation of `dot`:
```python
def dot_default(a, b, out=None):
    if ndim(a) == 0 or ndim(b) == 0:
        return a * b

    return a @ b


@create_numpy(_first2argreplacer, default=dot_default)
@all_of_type(ndarray)
def dot(a, b, out=None):
    return (a, b, out)
```

And then rewrite `multi_dot_default` as:
```python
def multi_dot_default(arrays):
    res = arrays[0]
    for a in arrays[1:]:
        res = dot(res, a)

    return res
```
A more sophisticated handling of shapes could also be done in `dot`.